src/solver.py

```diff
@@ -113,12 +122,51 @@ def solve_granddaughter(case: dict)->dict:
     if 'daughter_of_son_x2' in case:
         inh = 'daughter_of_son_x2'
         share = '2/3'
-    if 'son' in case or 'daughter_x2' in case:
+    if 'son' in case:
         case[inh] = 0
     elif 'son_of_son' in case:
         case[inh] = 'A'
+    elif 'daughter_x2' in case:
+        case[inh] = 0
     elif 'daughter' in case:
         case[inh] = '1/6'
     else:
         case[inh] = share
     return case
+
+def solve_asaba(case: dict,
+                rank: dict,
+                taseeb: dict) -> dict:
+    """
+    Solve for asaba inheritors except for the father/grandfather.
+
+    :param case:
+    :param fard_asaba:
+    :param rank:
+    :param taseeb: Dictionary containing the list of
+    inheritors who become asaba if this inheritor is present.
+    :return:
+    """
+    case_ranks = {inh: rank[inh] for inh in case if rank[inh] > 0}
+    if sisters_with_daughters(case=case):
+        if 'sister' in case:
+            case_ranks['sister'] = 5.5
+        if 'sister_x2' in case:
+            case_ranks['sister_x2'] = 5.5
+        if 'paternal_halfsister' in case:
+            case_ranks['paternal_halfsister'] = 6.5
+        if 'paternal_halfsister_x2' in case:
+            case_ranks['paternal_halfsister_x2'] = 6.5
+    if not case_ranks:
+        return case
```

Comment:
Not really a code comment, but just brainstorming. My guess is to resolve daughters and sisters case, we should probably go here right? Assuming we keep config file format as is

Reply:
Yes, this was a #todo just forgot to remove it.